src/main/resources/io/viash/platforms/nextflow/WorkflowHelper.nf

```diff
@@ -603,10 +606,467 @@ def paramsToList(params, config) {
+
+  // get the correct parser function for the detected params_list format
+  def paramListParsers = [ 
+    "csv": {[it, readCsv(it)]},
+    "json": {[it, readJson(it)]},
+    "yaml": {[it, readYaml(it)]},
+    "yaml_blob": {[null, readYamlBlob(it)]},
+    "asis": {[null, it]},
+    "none": {[null, [[:]]]}
+  ]
+  assert paramListParsers.containsKey(paramListFormat):
+    "Format of provided --param_list not recognised.\n" +
+    "You can use '--param_list_format' to manually specify the format.\n" +
+    "Found: '$paramListFormat'. Expected: one of 'csv', 'json', "+
+    "'yaml', 'yaml_blob', 'asis' or 'none'"
+  def paramListParser = paramListParsers.get(paramListFormat)
+
+  // fetch multi param inputs
+  def paramListOut = paramListParser(params.containsKey("param_list") ? params.param_list : "")
+  // multiFile is null if the value passed to param_list was not a file (e.g a blob)
+  // If the value was indeed a file, multiFile contains the location that file (used later).
+  def paramListFile = paramListOut[0]
+  def paramSets = paramListOut[1] // these are the actual parameters from reading the blob/file
+
+  // data checks
+  assert paramSets instanceof List: "--param_list should contain a list of maps"
+  for (value in paramSets) {
+    assert value instanceof Map: "--param_list should contain a list of maps"
+  }
+
+  // Reformat from List<Map> to List<Tuple2<String, Map>> by adding the ID as first element of a Tuple2
+  paramSets = paramSets.collect({ paramValues ->
+    [paramValues.get("id", null), paramValues.findAll{it.key != 'id'}]
+  })
+  // Split parameters with 'multiple: true'
+  paramSets = paramSets.collect({ paramSet ->
+    def id = paramSet[0]
+    def paramValues = paramSet[1]
+    def splitParamValues = _splitParams(paramValues, multiArguments)
+    [id, splitParamValues]
+  })
+  
+  // The paths of input files inside a param_list file may have been specified relatively to the
+  // location of the param_list file. These paths must be made absolute.
+  if (paramListFile){
+    paramSets = paramSets.collect({ paramSet ->
+      def id = paramSet[0]
+      def paramValues = paramSet[1]
+      def relativeParamValues = _resolvePathsRelativeTo(paramValues, inputfileArguments, paramListFile)
+      [id, relativeParamValues]
+    })
+  }
+
+  return paramSets
+}
+
+/**
+ * Cast parameters to the correct type as defined in the Viash config.
+ */
+
+private Map<String, Object> _castParamTypes(Map<String, Object> parValues, Map config) {
+  def configArguments = config.functionality.allArguments
+    // Cast the input to the correct type according to viash config
+  def castParValues = parValues.collectEntries({ parName, parValue ->
+    paramSettings = configArguments.find({it.plainName == parName})
+    // dont parse parameters like publish_dir ( in which case paramSettings = null)
+    parType = paramSettings ? paramSettings.get("type", null) : null
+    if (! (parValue instanceof Collection)) {
```

Comment:
It's good you put parentheses around the instanceof statement because indeed Groovy will otherwise interpret it as `(!parValue) instanceof Collection` which is incorrect. If you want, you can use `if (parValue !instanceof Collection) {` instead.